api/src/pcapi/core/offerers/api.py

```diff
@@ -223,22 +223,21 @@ def update_venue_location(
     On the other side, BO users might want to force a location to a venue, for example if the address is unknown
     for the API.
     """
-    if not any(field in modifications for field in ("street", "city", "postalCode")):
+    if not any(field in modifications for field in ("street", "city", "postalCode", "publicName", "name")):
         return
-
-    street = modifications.get("street") or venue.street
-    city = modifications.get("city") or venue.city
-    postal_code = modifications.get("postalCode") or venue.postalCode
-    latitude = modifications.get("latitude") or venue.latitude
-    longitude = modifications.get("longitude") or venue.longitude
-    ban_id = modifications.get("banId") or venue.banId
+    street = modifications.get("street") or venue.offererAddress.address.street  # type: ignore[union-attr]
+    city = modifications.get("city") or venue.offererAddress.address.city  # type: ignore[union-attr]
+    postal_code = modifications.get("postalCode") or venue.offererAddress.address.postalCode  # type: ignore[union-attr]
+    latitude = modifications.get("latitude") or venue.offererAddress.address.latitude  # type: ignore[union-attr]
+    longitude = modifications.get("longitude") or venue.offererAddress.address.longitude  # type: ignore[union-attr]
+    ban_id = modifications.get("banId") or venue.offererAddress.address.banId  # type: ignore[union-attr]
```

Comment:
Un peu dommage tous ces `type: ignore[union-attr]`. mypy se plaint parce que `venue.offererAddress` est nullable et que dans ce cas on ne pas à accéder à `address`. Or, avant de rentrer dans `update_venue_location`, on s'assure que `venue.isVirtual` est false. Du plus, on est plutôt que le script passé en amont ait fait son travail. On pourrait juste mettre un `assert venue.offererAddress is not None` ou `if venue.offererAddress: raise ...`.

Dans tous les cas, arriver ici avec `venue.offererAddress` lèvera une exception, donc autant retirer ces type ignore : )